msgraph/serviceprincipals_test.go

```diff
@@ -157,6 +176,70 @@ func testServicePrincipalsClient_ListGroupMemberships(t *testing.T, c ServicePri
 	if len(*groups) != 2 {
 		t.Fatalf("ServicePrincipalsClient.ListGroupMemberships(): expected groups length 2. was: %d", len(*groups))
 	}
+	return
+}
+
+func testServicePrincipalsClient_AddOwners(t *testing.T, c ServicePrincipalsClientTest, sp *msgraph.ServicePrincipal) {
+	status, err := c.client.AddOwners(c.connection.Context, sp)
+	if err != nil {
+		t.Fatalf("ServicePrincipalsClient.AddOwners(): %v", err)
+	}
+	if status < 200 || status >= 300 {
+		t.Fatalf("ServicePrincipalsClient.AddOwners(): invalid status: %d", status)
+	}
+}
+
+func testServicePrincipalsClient_ListOwnedObjects(t *testing.T, c ServicePrincipalsClientTest, id string) (ownedObjects *[]string) {
+	ownedObjects, _, err := c.client.ListOwnedObjects(c.connection.Context, id)
+	if err != nil {
+		t.Fatalf("ServicePrincipalsClient.ListOwnedObjects(): %v", err)
+	}
+
+	if ownedObjects == nil {
+		t.Fatal("ServicePrincipalsClient.ListOwnedObjects(): ownedObjects was nil")
+	}
+
+	if len(*ownedObjects) != 1 {
+		t.Fatalf("ServicePrincipalsClient.ListOwnedObjects(): expected ownedObjects length 1. was: %d", len(*ownedObjects))
+	}
+	return
+}
+
+func testServicePrincipalsClient_ListOwners(t *testing.T, c ServicePrincipalsClientTest, id string) (owners *[]string) {
+	owners, status, err := c.client.ListOwners(c.connection.Context, id)
+	if err != nil {
+		t.Fatalf("ServicePrincipalsClient.ListOwners(): %v", err)
+	}
+
+	if status < 200 || status >= 300 {
+		t.Fatalf("ServicePrincipalsClient.ListOwners(): invalid status: %d", status)
+	}
+
+	if len(*owners) != 1 {
+		t.Fatalf("ServicePrincipalsClient.ListOwners(): expected owners length 1. was: %d", len(*owners))
+	}
```

Comment:
```suggestion
	if len(*owners) == 0 {
		t.Fatalf("ServicePrincipalsClient.ListOwners(): expected at least 1 owner. has: %d", len(*owners))
	}
```
As I don't know the exact number of `expected` owners(your suggestion contains a non-existing `expected` variable), what would you say if we check as follows?

Reply:
My bad! I missed that when pasting. `expected` is an extra function arg - a string slice containing any expected owner UUIDs. The goal being to ensure at least a specific list of owners is correctly assigned.